### ISSUE TITLE
Agent removal command removes all agents in a pool

### ISSUE THREAD

**Issue by andydix** [MicrosoftDocs/vsts-docs#1744]
I'm using AWS EC2 instances in an Auto Scaling Group as "self-hosted" build/release agents. Each instance is self-installing and self-registering as they launch so I can have multiple agents. I've used the --agent switch and used the COMPUTERNAME so they are uniquely named. When one terminates and it uses a Lifecycle Hook to run a script which runs the "config.cmd remove" command to remove that agent from VSTS.

Trouble is, running the remove on one agent seems to remove **all** of them from my pool.

There doesn't seem to be the same --agent switch on the remove version of the command.

I am using the same PAT for them all which might be the issue (?) but surely the remove part of the command should have some knowledge of which agent is which and not just remove all of them?

Any help, gladly accepted.

Thanks
Andy

**Comment by TingluoHuang** [comment]
@andydix `config.cmd remove` should only unconfigure the agent you run `config.cmd remove`, can you share agent diag log with us? the log should under `_diag` folder from agent root directory.

**Comment by andydix** [comment]
Hi all,

I just spun up two agents and then manually ran the remove command on one of them. This only removed that agent i.e. correct behaviour.

I then reinstalled that agent manually and via the AWS console terminated one of the two instances. Both agents were removed from VSTS (soz, Azure DevOps 🤐 ).

When I look on the remaining instance, there's logging in _daig for agent removal!!

So I apologise for wasting your time. This is clearly an AWS issue as the Lifecycle Hook is being triggered for both instances despite only one of them actually being terminated. I will take this up with AWS.

Thanks for your replies and attention though.
